src/pointer-analysis/value_set.cpp

```diff
@@ -569,12 +569,32 @@ void value_sett::get_value_set_rec(
     // Find out what the pointer operand points at, and suck that data into
     // new object maps.
     object_mapt op0_set;
-    if(!is_pointer_type(op1))
-      get_value_set_rec(op0, op0_set, "", op0->type, false);
-
     object_mapt op1_set;
-    if(!is_pointer_type(op0))
-      get_value_set_rec(op1, op1_set, "", op1->type, false);
+    get_value_set_rec(op0, op0_set, "", (op0)->type, false);
+    get_value_set_rec(op1, op1_set, "", (op1)->type, false);
+
+    /*
+     * The main idea here is to be able to detect if any address might be valid at all.
```

Comment:
"The main idea here is to detect if any address might be valid at all. If any side of the pointer arithmetic is not a valid object, we can assume that it is not the pointer operand."